test/run-tests.py

```diff
@@ -615,28 +675,66 @@ def main(args):
     out_dir = tempfile.mkdtemp(prefix='sexpr-wasm-')
     out_dir_is_temp = True
 
+  variables['out_dir'] = os.path.abspath(out_dir)
+
+  allProcs = []
+  continuedErrors = 0
   try:
-    for i, p in enumerate(range(num_proc)):
-      args = (i, options, variables, inq, outq, out_dir)
-      proc = multiprocessing.Process(target=ProcessWorker, args=args)
-      proc.start()
+    if not options.no_multithreaded:
+      inq = multiprocessing.Queue()
+      outq = multiprocessing.Queue()
+      for info in infosToRun:
+        inq.put_nowait(info)
+      runWorkers = multiprocessing.Value('i', 1)
+      for i, p in enumerate(range(num_proc)):
+        args = (i, options, variables, inq, outq, out_dir, runWorkers)
+        proc = multiprocessing.Process(target=ProcessWorker, args=args)
+        allProcs.append(proc)
+        proc.start()
+      inq.close();
+
+      finished_tests = 0
+      while finished_tests < test_count:
+        try:
+          info, result = outq.get(True, 0.01)
+        except Queue.Empty:
+          status.UpdateTimer()
+          continue
 
-    finished_tests = 0
-    while finished_tests < test_count:
-      try:
-        info, result = outq.get(True, 0.01)
-      except Queue.Empty:
-        status.UpdateTimer()
-        continue
+        finished_tests += 1
+        HandleTestResult(status, info, result, options.rebase)
+        if options.stop and status.failed > continuedErrors:
+          shouldContinue = raw_input('Continue testing? (yes/no): ')
+          if len(shouldContinue) != 0 and shouldContinue[0] != 'y':
+            with runWorkers.get_lock():
+              runWorkers.value = 0
+            break
+          continuedErrors += 1
+    else:
+      for info in infosToRun:
+        result = RunTest(info, options, variables, out_dir)
+        HandleTestResult(status, info, result, options.rebase)
+        if status.failed > continuedErrors:
+          if options.stop:
+            rerunVerbose = raw_input('Rerun with verbose option? (yes/no): ')
+            if len(rerunVerbose) != 0 and rerunVerbose[0] != 'n':
+              RunTest(info, options, variables, out_dir, 2)
```

Comment:
last arg (`2`) is not clear without looking at the definition. Usually I make it explicit in that case:
```RunTest(info, options, variables, out_dir, verbose_level=2)```

Or maybe use constants? `VERBOSE_2` or something.